daemon/api_snapshots.go

```diff
@@ -173,3 +178,20 @@ func getSnapshotExport(c *Command, r *http.Request, user *auth.UserState) Respon
 
 	return &snapshotExportResponse{SnapshotExport: export}
 }
+
+func doSnapshotImport(c *Command, r *http.Request, user *auth.UserState) Response {
+
+	// XXX: check that we have enough space to import the compressed snapshots
+
+	// XXX: is this the right layer?
+	defer r.Body.Close()
```

Comment:
This looks like the right place, and inline with what we do elsewhere (otherwise Import would need to take ReadCloser). I'd remove this XXX comment.